skbio/tree/_tree.py

```diff
@@ -617,6 +617,77 @@ def lowest_common_ancestor(self, tipnames):
 
     lca = lowest_common_ancestor  # for convenience
 
+    def path(self, other, include_ends=False):
+        r"""Return the list of nodes in the path from one node to another.
+
+        Parameters
+        ----------
+        other : TreeNode
+            Final node of path.
+        include_ends: bool, optional
+            Whether to include the initial and final nodes in the list.
+
+        Returns
+        -------
+        list
+            List of TreeNode objects.
+
+        Examples
+        --------
+        >>> from skbio import TreeNode
+        >>> tree = TreeNode.read(["((a,b)c,(d,e)f)root;"])
+        >>> node_1, node_2 = tree.find('a'), tree.find('d')
+        >>> path = node_1.path(node_2)
+        >>> print(len(path))
+        3
+        >>> path_2 = node_1.path(node_2, include_ends=True)
+        >>> print(len(path_2))
+        5
+
+        """
+        # check to see if nodes are on same tree
+        root1, root2 = self.root(), other.root()
+        if root1 is not root2:
+            raise TypeError("Could not find path between nodes.")
+
+        # create list of ancestors for both nodes
+        anc1, anc2 = (
+            [node for node in self.ancestors()],
+            [node for node in other.ancestors()],
+        )
+
+        # find lowest common ancestor
+        for n1, n2 in zip(reversed(anc1), reversed(anc2)):
+            if n1 is n2:
+                lca = n1
+            else:
+                break
+
+        # delete nodes from intersection of ancestors
+        del anc1[anc1.index(lca) :]
```

Comment:
`del` is expensive because you are mutating a list in place. `index` is expensive too because you are searching a list, which is $O(n)$. I think a better way is to modify the above `for` loop into:

```python
for i, (n1, n2) in enumerate(zip(reversed(anc1), reversed(anc2))):
    ...
```

This will return the index of the lca without the need to find it in the subsequent code. Then you might do something like:

```python
path = anc1[:len(anc1) - i] + [n1] + anc2[:len(anc2) - i][::-1]
```